### PR TITLE
Fixed archived counts being counted in superscript.

### LINKED ISSUE #1459: Superscript should not count archived thoughts
## Steps to Reproduce

``` 
- =archive
  - a
- a
``` 

Do not toggle on `showHiddenThoughts`. `=archive` should be hidden. 

## Current Behavior

Superscript of `a` is 2. 

## Expected Behavior

`a` should not have a superscript. 

Archived thoughts should not count towards the superscript contexts when `showHiddenThoughts` is true.

### PR DESCRIPTION
**Pull request by ankitkarna99**
Fixes #1459

## Changes
- Removed `normalizeThought` function from `getAncesstorByValue` utility, since it removed the `=` from `=archive`, resulting in all archived thoughts being counted in the superscript.
- Fixed a typo `=archive` to `=archive` in `ThoughtAnnotation.native.tsx`

I checked all the use-cases of `getAncesstorByValue` and it seems to be used only to find meta values like `=archive`, so it seemed safe to remove the `normalizeThought` from the function.

### CODE REVIEW

File: src/components/ThoughtAnnotation.native.tsx [hunk 1]
```diff
@@ -150,7 +150,7 @@ const ThoughtAnnotation = ({
   /** Returns true if the thought is not archived. */
   const isNotArchive = (thoughtContext: ThoughtContext) =>
     // thoughtContext.context should never be undefined, but unfortunately I have personal thoughts in production with no context. I am not sure whether this was old data, or if it's still possible to encounter, so guard against undefined context for now.
-    showHiddenThoughts || !getAncestorByValue(state, thoughtContext, '=archieve')
+    showHiddenThoughts || !getAncestorByValue(state, thoughtContext, '=archive')
```
Comment (raineorshine) [inline src/components/ThoughtAnnotation.native.tsx:153]: 🤦‍♀️

**Review by raineorshine — CHANGES_REQUESTED** [review]
> I checked all the use-cases of `getAncesstorByValue` and it seems to be used only to find meta values like `=archive`, so it seemed safe to remove the `normalizeThought` from the function.

Indeed it is! Nice work.

Can you update the comment and variable names to remove references to `normalizedValue`? We always want to keep comments in line with the code.

File: src/components/ThoughtAnnotation.native.tsx (hunk 1 above)
Comment (ankitkarna99) [reply src/components/ThoughtAnnotation.native.tsx:153]: Thanks for reminding about the comments. They are updated now.

**Review by raineorshine — APPROVED** [review]
Perfect! Ready for unit test.

**Comment by ankitkarna99** [comment]
Which part of the code exactly should we unit test?

**Comment by raineorshine** [comment]
I think we should put the tests in `src/components/__tests__/Superscript.ts`, which doesn't exist yet. We'll need to test at the DOM level. I recommend using RTL for this. You can use https://github.com/cybersemics/em/blob/dev/src/components/__tests__/headings.ts as a starting point. You'll just need a way to select the superscript.

**Comment by ankitkarna99** [comment]
@raineorshine, I have added the test-cases for the superscript component using the React Testing Library. I could not find a way to select the `sup` tag with existing functions without adding the custom test id, so instead I wrote the test other way around. I found the superscript using `screen.getByText` and expected the element to be of type `sup`.

**Review by raineorshine — APPROVED** [review]
Looks great!

**Pull request merged**